bigtable/tableadmin/tableadmin.py

```diff
@@ -0,0 +1,253 @@
+
+    cf5 = table.column_family(column_family_id5, nested_rule)
+    cf5.create()
+    print 'Created column family cf5 with a Nested GC rule.'
+    # [END bigtable_create_family_gc_nested]
+
+    # [START Printing Column Family and GC Rule for all column families.]
+    print 'Printing Column Family and GC Rule for all column families...'
+    column_families = table.list_column_families()
+    for column_family, gc_rule in sorted(column_families.items()):
+        print "Column Family:", column_family
+        print "GC Rule:"
+        print gc_rule.to_pb()
+    # [END Printing Column Family and GC Rule for all column families.]
+
+
+def exists(instance_obj, table_id):
+    """ Check whether table exists or not.
+
+        Note: This is temporary function as code for this is under development.
+        Once complete, this function would be removed.
+
+    :type instance_obj: Instance Class.
+    :param instance_obj: Instance object.
+
+    :type table_id: str
+    :param table_id: Table id to create table.
+    Returns bool
+    """
+    for table in instance_obj.list_tables():
+        if table_id == table.table_id:
+            return True
+    return False
+
+
+def delete_table(project_id, instance_id, table_id):
+    ''' Delete bigtable.
+
+    :type project_id: str
+    :param project_id: Project id of the client.
+
+    :type instance_id: str
+    :param instance_id: Instance of the client.
+
+    :type table_id: str
+    :param table_id: Table id to create table.
+    '''
+
+    client = bigtable.Client(project=project_id, admin=True)
+    instance = client.instance(instance_id)
+    table = instance.table(table_id)
+
+    print 'Checking if table {} exists...'.format(table_id)
+    if exists(instance, table_id):
+        print 'Table {} exists.'.format(table_id)
+        print 'Deleting {} table.'.format(table_id)
+        table.delete()
+        print 'Deleted {} table.'.format(table_id)
+    else:
+        print 'Table {} does not exists.'.format(table_id)
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(
+        description=__doc__,
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+
+    parser.add_argument('command',
+                        help='run or delete. Operation to perform on table.')
+    parser.add_argument(
+        '--table',
+        help='Cloud Bigtable Table name.',
+        default='Hello-Bigtable')
+
+    parser.add_argument('project_id', help='Your Cloud Platform project ID.')
+    parser.add_argument(
+        'instance_id', help='ID of the Cloud Bigtable instance to connect to.')
+
+    args = parser.parse_args()
+
+    if args.command.lower() == 'run':
+        run_table_operations(args.project_id, args.instance_id, args.table)
+    elif args.command.lower() == 'delete':
+        delete_table(args.project_id, args.instance_id, args.table)
+    else:
+        print 'Command should be either run or delete.\n Use argument -h, \
+               --help to show help and exit.'

```

Comment:
Please remove the format characters here.

Reply:
Done